### ISSUE TITLE
Image Tracker blocked by uBlock Origin '1st-party scripts'

### ISSUE THREAD

**Issue by collinbarrett** [piwik/piwik#11311]
**tl;dr:**
I believe some tools like uBlock Origin flag the Image Tracker's "img" (really a php file) as a "script" due to the php file extension.

**Details:**
As I understand, the Image Tracker is to allow for bare-bones tracking of users who have javascript disabled. To test this, I tried loading my Piwik-enabled site in Chrome with uBlock Origin's "1st-party scripts" and "3rd-party scripts" blocked. (I use [Tracker Proxy](https://github.com/piwik/tracker-proxy), so blocking "1st-party scripts" is the rule of interest.)

However, I noticed that the noscript call to the "img src" of piwik.php is still blocked by uBlock Origin. (Note that I actually renamed "piwik.php" to "randomName.php" so that the keyword "piwik" is not blocked by any default uBlock Origin filter lists. I also verified that it was in fact the "1st-party scripts" rule that blocked it via uBlock Origin's logger.) It likely flags the php extension as a script as well.

Is there a way to replace the php file with an actual image file (jpg, gif, etc.) for noscript tracking so that uBlock Origin (and possibly other tools) cannot differentiate it as being a "script"? I understand I would have to adjust my caching rules so that the specific image is not cached at any level (browser, cdn, web server cache, etc.).

**Comment by mattab** [comment]
Hi @collinbarrett 

we haven't checked your findings. To answer your question we'd need to understand how ublock origin blocking policy works and why it would block the image tracker, and therefore how to work around this. My best bet would be to understand this first maybe by asking the ublock origin team or in their forums. This is for us out of scope. cheers

**Comment by negreanucalin** [comment]
The issue is that ad-blocking software checks each request and is a query parameter matches a given rule it block it. 
See: [My proposal](https://github.com/matomo-org/matomo/issues/14938)
